### ISSUE TITLE
no-undef not surpressed for instanceof

### ISSUE THREAD

**Issue by AlexDWu** [gajus/eslint-plugin-flowtype#163]
So I tried to use `instanceof` for a dynamic type check a la https://flowtype.org/docs/dynamic-type-tests.html#a-instanceof-b

so something like `target instanceof HTMLElement` and eslint was saying `HTMLElement` is undefined.

A quick search through this repository and eslint didn't seem to mention `instanceof`. Did I miss something?

**Comment by AlexDWu** [comment]
hmmm, added this to `VALID_WITH_DEFINE_FLOW_TYPE` in tests/rules/assertions/defineFlowType.js and it passed.

``` javascript
  {
    code: 'var a: AType; a instanceof AType',
    errors: [
      '\'AType\' is not defined.',
      '\'AType\' is not defined.'
    ]
  }
```

I must be missing something.

**Comment by AlexDWu** [comment]
hmm, I can't seem the reproduce this issue where i first saw it. oh well

**Comment by AlexDWu** [comment]
Hmm, this issue is showing up again for me. Help?

**Comment by AlexDWu** [comment]
Ah so because this is a dynamic check, the type that is being checked is being used as a value, and therefore needs to be defined somehow. Hmmm.
